src/engine/modules/velocity-module.ts

```diff
@@ -0,0 +1,198 @@
+
+  readonly ANGLE_OFFSET = -1.5708 // negative 90 degeress
+
+  private velocityX: VelocityDimension = new VelocityDimension();
+  private velocityY: VelocityDimension = new VelocityDimension();
+
+  private maxVelocityX: number;
+  private maxVelocityY: number;
+  private acceleration: number;
+  private deceleration: number;
+  private _angle: number = 0;
+
+  constructor(maxVelocity: number = 0, acceleration: number = 0, deceleration: number = 0) {
+    super();
+    this.maxVelocityX = maxVelocity;
+    this.maxVelocityY = maxVelocity;
+    this.acceleration = acceleration;
+    this.deceleration = deceleration;
+  }
+
+  update(gameObject: GameObject, deltaInMs: number): void {
+
+    var deltaInSeconds = deltaInMs / 1000;
+    this.velocityX.update(deltaInSeconds, this.maxVelocityX);
+    this.velocityY.update(deltaInSeconds, this.maxVelocityY);
+
+    gameObject.x += deltaInSeconds * this.velocityX.velocity;
+    gameObject.y += deltaInSeconds * this.velocityY.velocity;
+
+    // if (this.velocityX.velocity != 0 || this.velocityY.velocity != 0) {
+    //   this._angle = Math.atan2(this.velocityX.velocity, -this.velocityY.velocity) + this.ANGLE_OFFSET;
+    // }
+  }
+
+  get velX() {
+    return this.velocityX.velocity;
+  }
+
+  get velY() {
+    return this.velocityY.velocity;
+  }
+
+  get angle() {
+    return this._angle;
+  }
+
+  accelerateRight(): void {
+    this.accelerationX = this.acceleration;
+  }
+
+  accelerateLeft(): void {
+    this.accelerationX = -this.acceleration;
+  }
+
+  accelerateUp(): void {
+    this.accelerationY = -this.acceleration;
+  }
+
+  accelerateDown(): void {
+    this.accelerationY = this.acceleration;
+  }
+
+  decelerateRight(): void {
+    this.decelerationX = -this.deceleration;
+  }
+
+  decelerateLeft(): void {
+    this.decelerationX = this.deceleration;
+  }
+
+  decelerateUp(): void {
+    this.decelerationY = this.deceleration;
+  }
+
+  decelerateDown(): void {
+    this.decelerationY = -this.deceleration;
+  }
+
+  set accelerationX(pixelsPerSecond: number) {
+    this.velocityX.acceleration = pixelsPerSecond;
+    this.calculateAngle();
+  }
+
+  set accelerationY(pixelsPerSecond: number) {
+    this.velocityY.acceleration = pixelsPerSecond;
+    this.calculateAngle();
+  }
+
+  set decelerationX(pixelsPerSecond: number) {
+    this.velocityX.deceleration = pixelsPerSecond;
+  }
+
+  set decelerationY(pixelsPerSecond: number) {
+    this.velocityY.deceleration = pixelsPerSecond;
+  }
+
+  constantAccelerateAtAngleXY(acceleration: number, angle: number, speed: number) {
+    this.maxVelocityX = speed * Math.cos(angle);
+    this.maxVelocityY = speed * Math.sin(angle);
+    this.accelerationX = Math.abs(acceleration) * Math.cos(angle);
+    this.accelerationY = Math.abs(acceleration) * Math.sin(angle);
+    this._angle = angle;
+  }
+
+  // TODO: This needs improvement. Angle changes needs to be smoother.
+  private calculateAngle() {
+
+    var xFactor = 0;
```

Comment:
Gotta bunch of `var`s in this commit. Just do a find all and make these `let`. It's hard getting used to using let, but it's a lot nicer